int8[] target_x
int8[] target_y
float32[] obstacle_x
float32[] obstacle_y
